### PR TITLE
[Console] remove dead code

### PR DESCRIPTION
**Pull request by Tobion**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Part of #24961 for 2.7

### CODE REVIEW

**Comment by ro0NL** [comment]
ref #19813. Are we sure we dont re-introduce #18484 which somehow strongly suggests `FatalThrowableError`? (just some random observation :))

use + composer entry can be removed otherwise

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -112,15 +112,13 @@ public function run(InputInterface $input = null, OutputInterface $output = null
...
-        } catch (\Exception $x) {
-            $e = $x;
-        } catch (\Throwable $x) {
-            $e = new FatalThrowableError($x);
+        } catch (\Exception $e) {
+        } catch (\Throwable $e) {
         }
 
         if (null !== $e) {
-            if (!$this->catchExceptions || !$x instanceof \Exception) {
```
Comment (ro0NL) [inline src/Symfony/Component/Console/Application.php]: ah right, this causes $e being FatalThrowableError not to pass. Nice catch :-)

**Review by xabbuh — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by ro0NL — APPROVED** [review]
noise comments removed :) confused with other PR.

**Review by 20uf — APPROVED** [review]

**Comment by chalasr** [comment]
Good catch, thanks @Tobion.

**Comment by stof** [comment]
This is broken. You are now passing the Error to `renderException` instead of a FatalThrowableException, and this does not pass the typehint

**Comment by stof** [comment]
2.7 does not have a typehint, but 3.x does have it.

**Comment by chalasr** [comment]
@stof `Error` instances are rethrown , the `FatalThrowableError` instance was never passed to `renderException()`

**Pull request merged**